### PR TITLE
[test/AEM-5423] Create React Unit Test - CTA Button (including Icon Component) and Documentation

### PR DESCRIPTION
**Pull request by bpearsoncse**
https://jira.nationalgeographic.com/browse/AEM-5423

Of note: There are many errors showing up when running tests. These seem unrelated to these tests as I confirmed they are showing up on master. Seems related to Pestle, but I did not have time to investigate further. Also, I fixed a bug I discovered in testing in which not all styles were being applied correctly for some conditions.

### CODE REVIEW

File: test/app/modules/button/button_test.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        inverse={false}
+        onClick={() => {}}
+        onFocus={() => {}}
+        onBlur={() => {}}
+        style="default"
+        type="button"
+      />);
+    });
+
+    it('logs to console', () => {
```
Comment (krlospelaez) [inline test/app/modules/button/button_test.js]: Is this logs required?

**Comment by krlospelaez** [comment]
The errors are an expected behaviour though it would be better if they can be turned off manually.

File: test/app/modules/button/button_test.js (hunk 1 above)
Comment (bpearsoncse) [inline test/app/modules/button/button_test.js]: not required, just forgot to delete. deleting now.

**Comment by bpearsoncse** [comment]
back to you @xiwcx @krlospelaez

**Comment by krlospelaez** [comment]
LGTM :+1:

**Comment by xiwcx** [comment]
oh, i was just looking at this and @krlospelaez beat me to it. i agree LGTM

**Pull request merged**